### PR TITLE
Address review comments from Laura Cowen & Graham; Replace DockerHub with Azure Container Registry;

### PR DESCRIPTION
**Pull request by majguo**
This pull request consists of three major changes:
- Address review comments from Laura Cowen (see **Details** below);
- Address review comments from Graham about adding WebSphere Liberty support;
- Replace DockerHub with Azure Container Registry;

### Details

The link of original review comments is [here](https://ibm-cloud.slack.com/archives/G010BB252NA/p1596554033077900).

The details of change summary for Laura's review comments:

> Overall
>  The guide is quite long with lots of sections and sub-sections to navigate. Any way at all to make it shorter and appear simpler?

No, I don't have a good idea here.  :sweat_smile:

> Intro
>  The sentence “The foundational elements of technology in this article include the following.” - I would at least put a colon on the end (instead of a period) so that it introduces the list that follows. But I suggest rewording the sentence to “This scenario described in this articles includes the following technologies:” or something similar to be a bit more direct?

Changed by putting a colon on the end.

> Preprequisites section
>  “Finish the following prerequisites ” -> “Complete the following prerequisites”?

Changed.

> Not sure it’s necessary to say “per your needs”.

Removed.

> “get one for free for one year here.” -> “get an <Azure subscription free for one year>.” (put the link on the meaningful text rather than on a less meaningful “here”).

Changed.

> Set up …cluster section
> “The pull secret enables your Azure Red Hat OpenShift cluster to find the Open Liberty operator.” - should probably be “…Open Liberty Operator.” (upper-case O) as it is elsewhere.

Changed.

> Install the Open Liberty OpenShift Operator section
> “Install the Open Liberty Open Shift Operator” - “…OpenShift Operator” (one word for OpenShift).Is it possible to add a fine faint grey border around the screenshots to make it stand more crisply on the page (assuming you haven’t already and it’s just not rendering in github preview)? 

Good idea. Changed per your advice.

> “Fill out the values as shown in the following image. ” - I would end that in a colon instead of a period as it introduces the image that follows it.

Changed.

> From an accessibility perspective, it’d be good to spell out in text the values that they need to complete (not just referring them to the screenshot). Not just in terms of screenreaders but also for being able to read the image if your vision isn’t great.

Updated by adding the values in text that they need to complete.

> “and wait a minute or two.” - exactly what happens that indicates to them when they can continue to the next step? (eg is there a message displayed after a minute or two?).

Good catch. Updated per your advice.

> “Open Liberty operator” - upper-case O

Changed.

> Typo: “so it will can easily run the application” - should be “so it can easily run…“?

Good catch. Changed per your advice.

> Prepare the OL application section
> Officially “IBM WebSphere Traditional,” should be “IBM WebSphere Application Server”. Or at least lower-case the “traditional” as it’s not the actual product name.

Changed per your advice.

> “during the prerequisites” -> “in the Prerequisites section”?

Changed per your advice.

> “only Java EE 8 APIs ” - is it necessary to be in bold?

Bold style removed.

> “for build time…for runtime” -> “at build time… at run time”?

Changed per your advice.

> Run the application on Open Liberty section
> The intro is quite long (also contains some typos) before you get to the numbered steps and the bit about the server.xml effectively duplicates what’s in the first steps so that it’s not clear whether you should be doing something at this point or waiting until you get to step 1. Could the intro be shortened and include less detail; eg “To run the application on Open Liberty, you need to create an Open Liberty server configuration file so that the Liberty Maven plugin can package the application for deployment. The Liberty Maven plugin is not required to deploy the application to OpenShift but we will use it in this example with Open Liberty’s developer (dev) mode to easily run the application locally. Complete the following steps on your local computer:”

Changed per your advice.

> Maybe add an explanation to the end of each step. Step 1… “This server.xml configures the Open Liberty server with…“; Step 5… “…to start Open Liberty in dev mode.“; 

Changed per your advice.

> “You’ll see output similar to the followings in your console.” -> “You’ll see the following output in your console.”

Changed per your advice.

> Should the instructions about opening the browser and stopping the application be part of the numbered list as well?

Good idea. Changed per your advice.

> “Press Control-C to stop the application and Open Liberty server. Instead, you can use” - should that say “…. Alternatively, you can use…“? Why should they use one or the other? It would be clearer to just give a single instruction rather than making them choose when there doesn’t sem to be much between them?

The 2nd option is removed.

> I guess you need to run the application locally so that the relevant files are on the target directory ready to be containerised - is that correct? If so, should the section about running the app locally mention that? (so that the reader knows it’s not just something nice we’re asking them to do - it’s actually essential to the packaging).

No.  "mvn clean package" is also included as the 1st command to build application image. The major purpose of running application locally is to verify if the application works before containerizing the application.

> Deploy application on ARO 4 cluster section
> The intro to the Deploy application section should maybe say that you can deploy using either CLI or the GUI. Actually, when I edit blog posts on the OpenLiberty.io blog, I usually suggest only giving one set of instructions - blog posts aren’t usually meant to be comprehensive documentation but just the easiest/best way through the steps. If you were to remove one of the methods, it would shorten the guide a bit (it’s quite long with a lot of sections already). But if it’s important to understand Azure-specific steps of both methods, no worries.

Prefer to keep both, no changes.

> Build application image section
> “Here is the Dockerfile ” maybe change to something more active like “Edit the following Dockerfile for your application:” then the steps as you have them now.

No need to edit the provided Dockerfile. This section is removed and replaced with "Complete the following steps to build the application image:".

> Indent the commands (eg mvn clean package) bit so that it’s more clearly part of Step 2.

Changed per your advice.

> Run the application locally with Docker section
> “…to verify whether it works.” - end it with a colon to lead in to the numbered list?

Changed per your advice.

> Push the image section
> “push it to Docker Hub with these commands.” -> “push it to Docker Hub using the following commands:”

Changed per your advice.

> Prepare OpenLibertyApplication yaml file section
> I’m not sure what the reader is meant to do in this section. Are they meant to create the yaml file and place in the location mentioned? Are they meant to edit a yaml file that’s already there?

The reader is meant to edit a yaml file that's already there. Updated to express the intention.

### CODE REVIEW

**Review by lauracowen — APPROVED** [review]
LGTM

**Comment by majguo** [comment]
Will merge the PR. 
Reviewers, you can switch to target branch [jiangma-aro-openliberty-howto-public](https://github.com/edburns/azure-docs/tree/jiangma-aro-openliberty-howto-public) for further review if needed.
